src/DalService.ts

```diff
@@ -0,0 +1,26 @@
+import { AxiosResponse } from 'axios';
+import GenerateApiError from './utils/GenerateApiError';
+
+export default class DalService {
```

Comment:
Data Access Layer service? Can you just rename this to something like ChannelApeAPIResponseMapper to ensure it explains intent and avoids eventually being a catchall util class.